### ISSUE TITLE
warn on casting enum to numeric value

### ISSUE THREAD

**Issue by durka** [rust-lang/rust-clippy#542]
This code

``` rust
#![warn(clippy)]
#![warn(clippy_pedantic)]

fn it_works() {
    enum Foo { Bar = 300 }

    let x = 300i32 as u8;
    let y = Foo::Bar as u8;
}
```

hits a warning on `x` but not on `y`, even though both are truncated. Probably one of the `cast_*` lints should trigger whenever a repr(rust) enum is as-casted (or if e.g. a repr(i32) is casted to a u8), optionally checking to see if the enum actually has numeric variants that would cause problems.
